### ISSUE TITLE
[a11y audit] Veteran Facing Forms, VA Form 21-4138 - Post User Research (VFF Initial Audit)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#87424]
> [!IMPORTANT]
> - **Prototype:** [Staging Prototype](https://staging.va.gov/supporting-forms-for-claims/statement-to-support-claim-form-21-4138/introduction) 
> - **VFF:** Just completed [an audit, prior to user research ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/81979)c/o @jeana-adhoc 
> - **CAIA A11Y AUDIT:** Due to VFF having an A11Y were able to hold this ticket and audit for a future date, post updates from research and the VFF audit. We can now resume the audit.
> - **Resume Audit:** [as of July 8, 2024](https://dsva.slack.com/archives/C07ANLE3H6H/p1720466869302799) - if needed

##  CAIA Accessibility Review Prior to User Research

CAIA (@EvanAtCoforma and/or @mutiaadhocteam ) will review one prototype in staging for accessibility issues and make recommendations to the Veteran Facing Forms Team before upcoming user research:

### Prototype
[Prototype in Staging](https://staging.va.gov/supporting-forms-for-claims/statement-to-support-claim-form-21-4138/introduction)

### Accessibility Review

> [!TIP]
> - There are two options for this audit's deliverables a Google Sheet or a GitHub .md file. 
> - For screen-reader compatibility, we recommend using the markdown deliverable, linked below: 

#### Deliverable Options (select one)
- [ ] **Google Sheet:** [Accessibility complex audit review spreadsheet for VA Form 21-4138](https://docs.google.com/spreadsheets/d/1dbVrDyXDJxfdp3LztjLlND7x42WVrEH3RuN7fYYYMg0/edit?usp=sharing)
- [ ] **Markdown Deliverable:** [GitHub Markdown File](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/Deliverables/Veteran%20Facing%20Forms/2024%2007%2002%20Audit:%20VFF%20-%20VA%20Form%2021-4138.md)

### VFF Team Info

- Team name: Veteran Facing Forms
- OCTO-DE product owner: Matthew Dingee
- Product manager: Tricia Baker
- Designer: Rachel Shearer
- FE engineer: Jacob Penner, Robert Hasselle
- Product/team Slack channel: #veteran-facing-forms
- **A11Y**: Jeana Clark :accessibility 

## Definition of Done

```[tasklist]
### A11y Tasks to Complete
- [x] Review past audit(s) c/o VFF Team #81979
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/81979
- [x] Notification to resume audit - received 7/8/2024
- [x] Create accessibility review spreadsheet from template (also linked in ticket) - [Template for Complex A11Y Audit](https://docs.google.com/spreadsheets/d/1TO8dsDBqIFT-3lGmHtTdeetJzHMlqvbppVYAwYNPQh8/edit?usp=sharing)
- [x] Link to this ticket, and WIP artifact, in originating ticket
- [x] **OPTION 1:** (For screen reader compatibility use option 2) Complete the audit [artifact - Google Sheet](https://docs.google.com/spreadsheets/d/1dbVrDyXDJxfdp3LztjLlND7x42WVrEH3RuN7fYYYMg0/edit?usp=sharing)
- [x] **OPTION 2:** (Recommended as it is screen-reader compatible): Complete the [GitHub Markdown artifact](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/Deliverables/Veteran%20Facing%20Forms/2024%2007%2002%20Audit:%20VFF%20-%20VA%20Form%2021-4138.md)
```

```[tasklist]
### Deliverable
- [x] **Future Item:** Complete accessibility review
- [x] Notify team in comments and link to audit [added to comments](https://github.com/department-of-veterans-affairs/va.gov-team/issues/87424#issuecomment-2293576463)
- [x] Notify team in Slack  [Ref. Slack Convo 8/16/2024](https://dsva.slack.com/archives/C07H1GQ86CE/p1723813499007909)
```

**Comment by sara-amanda** [comment]
## VFF Audit - Completed

@EvanAtCoforma - while you were OOO, [Jeana completed her audit](https://dsva.slack.com/archives/C07ANLE3H6H/p1720466869302799), linked in the task list in this ticket. 

### CAIA Audit TBD

Let me know after reviewing if you would like to keep this ticket open - or close it. 

cc: @NaomiPMC

**Comment by sara-amanda** [comment]
## Limited GitHub Access

Tagging @mutiaadhocteam in this ticket, until we can add Maria successfully to the assignees in this ticket, when access is provided, during the onboarding process. 

cc: @NaomiPMC @EvanAtCoforma

**Comment by sara-amanda** [comment]
# a11y Feedback

> [!TIP]
> This feedback is the result of the staging prototype being reviewed by an experienced, native screen reader, and it details their findings from their full, authentic experience. 
> - Therefore, we understand that some of the findings in this feedback may be existing issues on file with VA, and could be undergoing, or have, pending, planned remediation for post-launch. 
> - If that is the case, we understand, completely, so feel free to review and consider, only the applicable feedback. 
> - CAIA a11y feedback is not currently mandated, and is meant to be a resource for VFS teams, in order to create an equitable experience for our users. We hope that some, all or most of this feedback can be helpful now, or in the future, and generate a collaborative discussion. 
> - Should you have any questions, always feel free to reach out. 

## Testing Details
- **AT**:  NVDA
- **Device and OS**: Windows Desktop
- **Browser**: Chrome
- **Tester**: Maria 
    - **GitHub**: @mutiaadhocteam 
    - **Slack**:  [@Maria Mutia](https://dsva.slack.com/archives/D07A6N82TEH)
- **Feedback Shared**: 8/15/2024
- **Additional Testing Notes:** Reviewed the DOM, so the issues should be easily replicated across different browsers and platforms.

## Finding 1: "Continue" button is located at the end of the screen in VA Form 21-4138

<details><summary>Toggle to access findings and recommendations</summary>

- **Screen**: All pages with "Continue" buttons
- **Version**: All
- **Experience standard:** [Findability](https://design.va.gov/about/experience-standards/#category-findability)
- **Description**: The "Continue" button is located at the end of the screens throughout the form.
- **Details**: 
    - Users may have difficulty determining the next steps for the form.
    - Additionally, users may find it difficult to navigate through a considerable amount of content (form fields, dropdowns, text, etc.) before locating the "Continue" button.
  
<img width="960" alt="vetform2" src="https://github.com/user-attachments/assets/2e602d1d-d289-40f2-a666-374d3557b3e8">

### Recommendation for Finding 1
The "Continue" button should be located next to the "Back" link.

</details>

### VFS Team Feedback for Finding 1
Feel free to share the status of this finding, by checking the appropriate box, below. 
- [ ] Reviewed Feedback
- [ ] Known Issue
- [ ] Future Item


## Finding 2: No success message automatically announced upon updating information

<details><summary>Toggle to access findings and recommendations</summary>

- **Screen**: [6 of 6 Review application](https://staging.va.gov/supporting-forms-for-claims/statement-to-support-claim-form-21-4138/review-and-submit)
Version: All
- **Experience standard**:  [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
- **Description**: No success message is automatically announced upon updating information
- **Details**: 
    - Users may have difficulty understanding that the information on the page has been successfully updated.
    - Focus is sent back to the "Edit" button on updating information in a section.
    - To determine that the information has been updated, the user would have to navigate through the content. Users may find this to be disruptive and cumbersome.

<img width="960" alt="vetform1" src="https://github.com/user-attachments/assets/738a334d-987e-46b7-98cf-e771befdd0a0">

### Recommendation for Finding 2
A success message should be automatically announced upon updating or otherwise editing information on the screen.

</details>

### VFS Team Feedback for Finding 2
Feel free to share the status of this finding, by checking the appropriate box, below. 
- [ ] Reviewed Feedback
- [ ] Known Issue
- [ ] Future Item

## Finding 3: No error message automatically announced upon submitting an application containing errors

<details><summary>Toggle to access findings and recommendations</summary>

- **Screen**: [6 of 6 Review application](https://staging.va.gov/supporting-forms-for-claims/statement-to-support-claim-form-21-4138/review-and-submit)
- **Version**: All
- **Experience standard**: [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
- **Description**: No error message is automatically announced upon submitting an application containing errors.
- **Details**:
    - Users may have difficulty understanding than an error is preventing them from submitting the form.
    - Users will have to navigate through the content on the screen to understand that there is an error on the page.

<img width="960" alt="vetform3" src="https://github.com/user-attachments/assets/620948f4-eb38-4373-82e6-1d802fa75db5">

### Recommendation for Finding 3
An error message should be announced upon submitting a form with errors. This will inform the user that an error exists on the page and will prevent the user from unnecessary navigation of the page's content.

</details>

### VFS Team Feedback for Finding 3
Feel free to share the status of this finding, by checking the appropriate box, below. 
- [ ] Reviewed Feedback
- [ ] Known Issue
- [ ] Future Item

## Finding 4: Focus is not automatically sent to the location of an error

<details><summary>Toggle to access findings and recommendations</summary>

- **Screen**: [6 of 6 Review application](https://staging.va.gov/supporting-forms-for-claims/statement-to-support-claim-form-21-4138/review-and-submit)
Version: All
- **Experience standard**: [Findability](https://design.va.gov/about/experience-standards/#category-findability)
- **Description**: Focus is not sent to the error that exists on the page upon submission of an application that contains an error.
- **Details**:
    - Users may have difficulty understanding where an error exists on the page.
    - Users will have to navigate through the page's content to locate the error.

### Recommendation for Finding 4
Focus should be sent to the error that exists on the page.

</details>

### VFS Team Feedback for Finding 4
Feel free to share the status of this finding, by checking the appropriate box, below. 
- [ ] Reviewed Feedback
- [ ] Known Issue
- [ ] Future Item

## Next Steps
- [ ] Review Findings
- [ ] Mark status under each issue
- [ ] Reach out to CAIA with any questions

**Comment by sara-amanda** [comment]
> [!NOTE]
> ## CAIA INTERNAL
> - [x] This ticket is linked to its parent research ticket.
> - [x] This work has been completed
> - [x] Closing ticket
> - [x] Noted in Slack to team 12/17/2024, [the potential to check the boxes for findings 1-4](https://dsva.slack.com/archives/C07H1GQ86CE/p1734463121839679), to inform future research, relating to if the finding was: `Reviewed Feedback`, `Known Issue`, or `Future Item`
> - [x] Adding to tracker for CAIA Sprint 140
